boards/PDM/Src/Io/Io_VoltageCheck.c

```diff
@@ -0,0 +1,63 @@
+#include "App_InRangeCheck.h"
+#include "Io_VoltageCheck.h"
+
+static struct PdmCanTxInterface *_can_tx = NULL;
+
+void Io_VoltageInRangeCheck_Init(struct PdmCanTxInterface *can_tx)
+{
+    _can_tx = can_tx;
+}
+
+float Io_VoltageInRangeCheck_GetVbatVoltage(void)
+{
+    // TODO: Fix hard-coded value once the ADC is configured
+    return 7.0f;
+}
+
+void Io_VoltageInRangeCheck_VbatErrorCallback(enum InRangeCheck_Status status)
+{
+    if (status == VALUE_UNDERFLOW)
+    {
+        App_CanTx_SetPeriodicSignal_UNDERVOLTAGE_VBAT(_can_tx, true);
+    }
+    else if (status == VALUE_OVERFLOW)
+    {
+        App_CanTx_SetPeriodicSignal_OVERVOLTAGE_VBAT(_can_tx, true);
+    }
+}
+
+float Io_VoltageInRangeCheck_Get24vAuxVoltage(void)
+{
+    // TODO: Fix hard-coded value once the ADC is configured
+    return 23.0f;
+}
+
+void Io_VoltageInRangeCheck_24vAuxErrorCallback(enum InRangeCheck_Status status)
+{
+    if (status == VALUE_UNDERFLOW)
+    {
+        App_CanTx_SetPeriodicSignal_UNDERVOLTAGE_24_V_AUX(_can_tx, true);
+    }
+    else if (status == VALUE_OVERFLOW)
+    {
+        App_CanTx_SetPeriodicSignal_OVERVOLTAGE_24_V_AUX(_can_tx, true);
+    }
+}
+
+float Io_VoltageInRangeCheck_Get24vAccVoltage(void)
+{
+    // TODO: Fix hard-coded value once the ADC is configured
+    return 23.0f;
+}
+
+void Io_VoltageInRangeCheck_24vAccErrorCallback(enum InRangeCheck_Status status)
+{
+    if (status == VALUE_UNDERFLOW)
+    {
+        App_CanTx_SetPeriodicSignal_UNDERVOLTAGE_24_V_ACC(_can_tx, true);
+    }
+    else if (status == VALUE_OVERFLOW)
+    {
+        App_CanTx_SetPeriodicSignal_OVERVOLTAGE_24_V_ACC(_can_tx, true);
+    }
+}
```

Comment:
Another thought I had here, it doesn't seem like it would be too hard to make a quick macro to reduce the amount of duplication in this file. Same goes for the current checks. Something like:
```
#define CHOOSE_A_BETTER_MACRO_NAME(NAME1, NAME2) \
void Io_VoltageInRangeCheck_##NAME1##ErrorCallback(enum InRangeCheck_Status status) \
{ \
    if (status == VALUE_UNDERFLOW) \
    { \
        App_CanTx_SetPeriodicSignal_UNDERVOLTAGE_##NAME2(_can_tx, true); \
    } \
    else if (status == VALUE_OVERFLOW) \
    { \
        App_CanTx_SetPeriodicSignal_OVERVOLTAGE_##NAME2(_can_tx, true); \
    } \
}
```
It's not pretty, but  pretty effective for reducing duplication here. 

Anyhow, totally your call.

Reply:
Oh yeah I would prefer this. I was getting tired of copying-and-pasting for sure.

Reply:
Went for a static function instead :v:

Reply:
Nice, that's better then what I suggested.